### ISSUE TITLE
yumemi not working (rhash dlopen)

### ISSUE THREAD

**Issue by tamwile** [fpob/yumemi#3]
`doas pacman -S rhash`

```doas pacman -S rhash
doas (user@host) password: 
warning: rhash-1.4.4-1 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) rhash-1.4.4-1

Total Download Size:   0.17 MiB
Total Installed Size:  0.32 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n] 
:: Retrieving packages...
 rhash-1.4.4-1-x86_64                                                                                                172.9 KiB  2.11 MiB/s 00:00 [########################################################################################] 100%
(1/1) checking keys in keyring                                                                                                                   [########################################################################################] 100%
(1/1) checking package integrity                                                                                                                 [########################################################################################] 100%
(1/1) loading package files                                                                                                                      [########################################################################################] 100%
(1/1) checking for file conflicts                                                                                                                [########################################################################################] 100%
(1/1) checking available disk space                                                                                                              [########################################################################################] 100%
:: Processing package changes...
(1/1) reinstalling rhash                                                                                                                         [########################################################################################] 100%
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
```

```python
Traceback (most recent call last):
  File "/home/user/.local/bin/yumemi", line 5, in <module>
    from yumemi.cli import main
  File "/home/user/.local/pipx/venvs/yumemi/lib/python3.11/site-packages/yumemi/cli.py", line 7, in <module>
    import rhash
  File "/home/user/.local/pipx/venvs/yumemi/lib/python3.11/site-packages/rhash/__init__.py", line 16, in <module>
    from .rhash import (
  File "/home/user/.local/pipx/venvs/yumemi/lib/python3.11/site-packages/rhash/rhash.py", line 93, in <module>
    _LIBRHASH = CDLL(_LIBNAME)
                ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/ctypes/__init__.py", line 376, in __init__
    self._handle = _dlopen(self._name, mode)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: librhash.so.0: cannot open shared object file: No such file or directory
```

And when i copy and eval this in the python interpreter:
```python
>>> import sys
>>> from ctypes.util import find_library
>>> 
>>> 
>>> _LIBNAME = find_library('rhash')
>>> 
>>> if not _LIBNAME:
...     if sys.platform == "win32":
...         _LIBNAME = "librhash.dll"
...     elif sys.platform == "darwin":
...         _LIBNAME = "librhash.1.dylib"
...     elif sys.platform == "cygwin":
...         _LIBNAME = "cygrhash.dll"
...     elif sys.platform == "msys":
...         _LIBNAME = "msys-rhash.dll"
...     else:
...         _LIBNAME = "librhash.so.1"
... 
>>> _LIBNAME
'librhash.so.1'
>>> print(sys.platform)
linux
>>> from ctypes import CDLL
>>> _LIBRHASH = CDLL(_LIBNAME)
>>> _LIBRHASH
<CDLL 'librhash.so.1', handle 55ddab2fefa0 at 0x7ffb51b25f90>
```

Why is it librhash.so.1 in the interpreter and librhash.so.0 when we run yumemi ?

**Comment by fpob** [comment]
It is incompatibility between librhash >=1.4.4 and Python bindings from PyPI. I noticed this a while ago and added patched bindings that works with both versions of librhash, but the fixed yumemi is not released yet.

**Comment by fpob** [comment]
Just released 2.0.2, update and try

**Comment by tamwile** [comment]
Seems to work, thanks.
